### ISSUE TITLE
请问你训练用了多长时间？谢谢

### ISSUE THREAD

**Issue by ZHAIXINGZHAIYUE** [Tonyfy/LCNN_TRAIN#5]

**Comment by Tonyfy** [comment]
@ZHAIXINGZHAIYUE  一周左右。

**Comment by Tonyfy** [comment]
@ZHAIXINGZHAIYUE 看你的显卡怎么样了，其实我的loss还处于下降状态，不过后面没让它训练了。

**Comment by ZHAIXINGZHAIYUE** [comment]
还有一个问题要请教，你结束训练时的loss降到了多少？

**Comment by Tonyfy** [comment]
2左右

**Comment by iscas-lee** [comment]
我使用@Tonyfy的  solve文件，发现训练不收敛，请教是怎么回事呢？

net: "c:/caffe/caffe-windows-cuDNNv5/examples/deepface/LCNN_train_test.prototxt"
test_iter: 100
test_interval: 1000
test_compute_loss: true

base_lr: 0.001
momentum: 0.9
weight_decay: 0.0005
lr_policy: "inv"
gamma: 0.000005
power: 0.75

display: 100
max_iter: 3000000
snapshot: 10000
snapshot_prefix: "c:/caffe/caffe-windows-cuDNNv5/examples/deepface/10575/LCNN"
solver_mode: GPU

debug_info: false

I0921 09:07:34.047699  6920 solver.cpp:391] Test loss: 1.15091
I0921 09:07:34.047699  6920 solver.cpp:404]     Test net output #0: accuracy = 0.815
I0921 09:07:34.094699  6920 solver.cpp:228] Iteration 979000, loss = 1.26092
I0921 09:07:34.094699  6920 solver.cpp:244]     Train net output #0: accuracy = 0.71875
I0921 09:07:34.094699  6920 sgd_solver.cpp:106] Iteration 979000, lr = 0.000264324
I0921 09:07:46.331699  6920 solver.cpp:228] Iteration 979100, loss = 0.852175
I0921 09:07:46.331699  6920 solver.cpp:244]     Train net output #0: accuracy = 0.796875
I0921 09:07:46.331699  6920 sgd_solver.cpp:106] Iteration 979100, lr = 0.000264308
I0921 09:07:58.497699  6920 solver.cpp:228] Iteration 979200, loss = 1.23714
I0921 09:07:58.497699  6920 solver.cpp:244]     Train net output #0: accuracy = 0.6875
I0921 09:07:58.497699  6920 sgd_solver.cpp:106] Iteration 979200, lr = 0.000264291
I0921 09:08:10.666699  6920 solver.cpp:228] Iteration 979300, loss = 0.631258
I0921 09:08:10.666699  6920 solver.cpp:244]     Train net output #0: accuracy = 0.84375
I0921 09:08:10.666699  6920 sgd_solver.cpp:106] Iteration 979300, lr = 0.000264274
I0921 09:08:22.839699  6920 solver.cpp:228] Iteration 979400, loss = 0.956994
I0921 09:08:22.839699  6920 solver.cpp:244]     Train net output #0: accuracy = 0.75
I0921 09:08:22.839699  6920 sgd_solver.cpp:106] Iteration 979400, lr = 0.000264257
I0921 09:08:35.066900  6920 solver.cpp:228] Iteration 979500, loss = 1.25873
I0921 09:08:35.066900  6920 solver.cpp:244]     Train net output #0: accuracy = 0.75
I0921 09:08:35.066900  6920 sgd_solver.cpp:106] Iteration 979500, lr = 0.00026424
I0921 09:08:47.233899  6920 solver.cpp:228] Iteration 979600, loss = 1.57497
I0921 09:08:47.233899  6920 solver.cpp:244]     Train net output #0: accuracy = 0.75
I0921 09:08:47.233899  6920 sgd_solver.cpp:106] Iteration 979600, lr = 0.000264223
I0921 09:08:59.395400  6920 solver.cpp:228] Iteration 979700, loss = 0.930002
I0921 09:08:59.395400  6920 solver.cpp:244]     Train net output #0: accuracy = 0.765625
I0921 09:08:59.395400  6920 sgd_solver.cpp:106] Iteration 979700, lr = 0.000264207
I0921 09:09:11.558899  6920 solver.cpp:228] Iteration 979800, loss = 1.32372
I0921 09:09:11.558899  6920 solver.cpp:244]     Train net output #0: accuracy = 0.75
I0921 09:09:11.558899  6920 sgd_solver.cpp:106] Iteration 979800, lr = 0.00026419
I0921 09:09:23.717900  6920 solver.cpp:228] Iteration 979900, loss = 1.29074
I0921 09:09:23.717900  6920 solver.cpp:244]     Train net output #0: accuracy = 0.65625
I0921 09:09:23.717900  6920 sgd_solver.cpp:106] Iteration 979900, lr = 0.000264173

**Comment by Tonyfy** [comment]
@iscas-lee 不要停，继续train.

**Comment by getengqing** [comment]
@iscas-lee 发现我的accuracy=0,这个是哪里出问题了呢？
I1214 15:11:57.118721  4846 solver.cpp:228] Iteration 14800, loss = 9.21328
I1214 15:11:57.118752  4846 solver.cpp:244]     Train net output #0: accuracy = 0
I1214 15:11:57.118760  4846 sgd_solver.cpp:106] Iteration 14800, lr = 0.000947866
I1214 15:12:03.002496  4846 solver.cpp:228] Iteration 14900, loss = 9.2782
I1214 15:12:03.002529  4846 solver.cpp:244]     Train net output #0: accuracy = 0
I1214 15:12:03.002539  4846 sgd_solver.cpp:106] Iteration 14900, lr = 0.000947535
I1214 15:12:08.838096  4846 solver.cpp:337] Iteration 15000, Testing net (#0)
I1214 15:12:09.323823  4846 solver.cpp:391] Test loss: 9.15512

**Comment by tiankong1993** [comment]
我跟你出现了一样的问题，请问你解决了吗 @getengqing

**Comment by getengqing** [comment]
@tiankong1993 继续训练，不要停，迭代多一点步数就好

**Comment by iscas-lee** [comment]
@getengqing 
我的训练了很久很久，估计有半个月以上了，accuracy 始终也不高，最高的好像有个90%就上不去了。loss 基本处于不变的状态。
不知道大家有没有类似的问题。

**Comment by tiankong1993** [comment]
那你准确率90%的时候在lfw验证的效果怎么样，那个博主说他准确率40%在lfw上效果就不错了

**Comment by getengqing** [comment]
@iscas-lee 感觉还好，后面就没有训练了
I1215 08:46:12.730224  6640 solver.cpp:228] Iteration 968100, loss = 0.322078
I1215 08:46:12.730252  6640 solver.cpp:244]     Train net output #0: accuracy = 0.9375
I1215 08:46:12.730259  6640 sgd_solver.cpp:106] Iteration 968100, lr = 0.000266172
I1215 08:46:18.821393  6640 solver.cpp:228] Iteration 968200, loss = 0.744603
I1215 08:46:18.821424  6640 solver.cpp:244]     Train net output #0: accuracy = 0.9375

**Comment by Tonyfy** [comment]
这里的accuracy是分类的准确率（不是在lfw上人脸验证的准确率），有80%其实已经收敛了，就不用在训练下去了。可以监测一下train 的loss变化啊。

**Comment by getengqing** [comment]
@Tonyfy 感谢分享LCNN代码，能不能将lfw上人脸验证准确率的脚本一起提供！3Q
